c2rust-analyze/scripts/auto_fix_errors.py

```diff
@@ -0,0 +1,286 @@
+import argparse
+from collections import namedtuple
+import json
+import re
+import sys
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(
+        description='automatically apply rustc-suggested fixes for compile errors')
+    parser.add_argument('-n', '--dry-run', action='store_true',
+        help="print fixes that would be applied, but don't modify any files")
+    parser.add_argument('path', metavar='ERRORS.JSON',
+        help='output of `rustc --error-format json`')
+    return parser.parse_args()
+
+Fix = namedtuple('Fix', (
+    # Path to the file to modify.
+    'file_path',
+    # Line number where the fix will be applied.  This is used for debug output
+    # only.
+    'line_number',
+    # Start of the byte range to replace.
+    'start_byte',
+    # End (exclusive) of the byte range to replace.
+    'end_byte',
+    # Replacement text (as a `str`, not `bytes`).
+    'new_text',
+    # The original error message.  This is used for debug output only.
+    'message',
+))
+
+LifetimeBound = namedtuple('LifetimeBound', (
+    'file_path',
+    'line_number',
+    # Byte offset of the start of the lifetime parameter declaration.
+    'start_byte',
+    # Byte offset of the end of the lifetime parameter declaration.
+    'end_byte',
+    # The lifetime to use in the new bound.  If `'a: 'b` is the suggested
+    # bound, then `start/end_byte` points to the declaration of `'a`, and
+    # `bound_lifetime` is the string `"'b"`.
+    'bound_lifetime',
+))
+
+RemoveDeriveCopy = namedtuple('RemoveDeriveCopy', (
+    'file_path',
+    'line_number',
+    # Byte offset of the start of the token `Copy`.
+    'start_byte',
+    # Byte offset of the end of the token `Copy`.
+    'end_byte',
+))
```

Comment:
nit: I think it's a lot more idiomatic to use `@dataclass`es instead of `namedtuple`.

Reply:
I originally used `@dataclass` here, but switched to `namedtuple` because CI was failing - the Ubuntu 18 runner has Python 3.6, but dataclasses were added in 3.7

Reply:
Oh that's really annoying. Python 3.6 was EOL 3 years ago.